web/resources/analysis.py

```diff
@@ -106,25 +107,46 @@ def get(self, analysis_id):
 	def put(self, analysis_id, **kwargs):
 		analysis = utils.first_or_404(
 			Analysis.query.filter_by(hash_id=analysis_id))
-		if analysis.locked is True:
+		if analysis.status != 'DRAFT':
 			utils.abort(422, "Analysis is not editable. Try cloning it.")
-		elif 'locked' in kwargs:
-			if kwargs['locked'] > analysis.locked:
-				kwargs['locked_at'] = datetime.datetime.utcnow()
-				### Add other triggers here
 		return put_record(db.session, kwargs, analysis)
 
+	@doc(summary='Delete analysis.')
+	@utils.auth_required
```

Comment:
Many routes need to be restricted to ensure that users can only edit/delete their own analyses